draft-pauly-taps-arch.md

```diff
@@ -339,6 +339,56 @@ The Transport System Implementation Concepts define the set of objects used inte
 
 * Endpoint Racing: Endpoint Racing is the act of attempting to establish, or scheduling attempts to establish, multiple Protocol Stacks that differ based on the specific representation of the Remote Endpoint and the Local Endpoint, such as IP addresses resolved from a DNS hostname.
 
+### Message Framing, Parsing, and Serialization {#framing}
+
+While some transports expose a byte stream abstraction, most higher level
+protocols impose some structure onto that byte stream. That is, the higher
+level protocol operates in terms of messages, protocol data units (PDUs),
+rather than using unstructured sequences of bytes, with each message being
+processed in turn.  Protocols are specified in terms of state machines
+acting on semantic messages, with parsing the byte stream into messages
+being a necessary annoyance, rather than a semantic concern.  Accordingly,
+the Transport Services architecture exposes messages as the primary 
+abstraction.  Protocols that deal only in byte streams, such as TCP,
+represent their data in each direction as a single, long message.  When
```

Comment:
Just for the record, I don't have a problem with the "TCP represents their data as a single long message" statement per se (although I don't think it adds much value to say this) - my issue is when this translates into "thus, sending 50 small messages may mean 50 small TCP connections".  Obviously we don't want that.

Anyway - for this PR, this is fine by me!

Reply:
50 small messages *does not* mean 50 small TCP connections, and I am not advocating that anyone does that. =)

Reply:
I thought it does, in your system, if I don't care about using framers and just use "send" ?   Anyway sure we all agree that nobody wants 50 short TCP connections  :)

Reply:
Either way: this text is also fine to keep. So there's only one small item left below that is a change request - the "must" / "should"